code/modules/lighting/lighting_system.dm

```diff
@@ -246,10 +246,27 @@
 	anchored = 1
 
 /atom/movable/light/Destroy(force)
-	if(force)
-		. = ..()
-	else
+	if(!force)
 		return QDEL_HINT_LETMELIVE
+	var/turf/T = loc
+	. = ..()
+	if (T.lighting_object == src)
+		T.lighting_object = null
+	
+/atom/movable/light/New()
+	if (!loc || !isturf(loc))
```

Comment:
don't worry, unlike my comment suggests, I wasn't actually even confident it produced the same results! I had to actually write out those two "X, it succeeds" sentences to prove it to myself haha, that's half 2 in the morning for you :S